### ISSUE TITLE
Add-ons server inaccessible with MSVC build

### ISSUE THREAD

**Issue by Wedge009** [wesnoth/wesnoth#6171]
### Game and System Information
Self-compiled 1.15.18+dev (9c2ad49026282107c3c2e35c595a41ac0e86798f) and 1.17.0+dev (e49b5c7b0e18944bbcc6e21339e2269ec29dff26) on Windows 7.

### Describe the bug
Attempting to access the add-ons server results in a 'Remote host disconnected' error message. Logs report:
> error network: network_asio::error thrown during transaction with add-on server; "The handle is invalid"
error general: Remote host disconnected.

Attempting to access the add-ons server a second time results in a crash...

In contrast, there are no issues with logging in to the MP server. Accessing the add-ons server is fine on 1.14.17 release on Windows 7. No such issues on Linux.

### To Reproduce
1. On Windows 7 start Wesnoth 1.15+.
2. Click Add-ons in the main menu.
3. Click Connect.
4. Repeating steps 2 and 3 results in a crash.

### Expected behaviour
List of add-ons for the running Wesnoth version is downloaded and then displayed.

### Additional context
https://forums.wesnoth.org/viewtopic.php?t=54955 may be related as it has the same symptoms but the exception message is different. Like the reporter there, accessing the web page from Internet Explorer is fine.

**Comment by Pentarctagon** [comment]
Does it happen with the Steam or SourceForge versions, since those are compiled in a different way (assuming you're compiling with VS yourself)?

**Comment by Wedge009** [comment]
I just retried using 1.15.18 win64 downloaded from SourceForge. Strange, seems to work just fine. I'll try a full recompile, it was working fine previously...

(Yeah, self-compiled in VS2019.)

As an aside, I confirmed #4508 still happens with the official build (it's using SDL 2.0.12) so that one definitely isn't me doing something wrong/different.

**Comment by Pentarctagon** [comment]
The official Windows builds don't use VS, they're cross-compiled from Linux, so that's a big difference (even though in a perfect world they'd be entirely similar).

**Comment by Wedge009** [comment]
Yes, I'm aware of that - or at least that they aren't built from MSVC.

**Comment by Pentarctagon** [comment]
Alright.  Are you able to check if this happens on Windows 10 as well?  If it's just Windows 7 then I'd be fine with this staying closed, but if it also happens on Windows 10 it'd be good to keep it open given that building with VS is the primary way Windows devs are expected to build Wesnoth.

**Comment by Wedge009** [comment]
@Vultraz I saw you mention that you're already using VS2022 - can you advise if you have any issues accessing the Add-ons server with your builds? If not, it suggests an issue with my boost set-up or something else specific to my Windows 7 environment.

**Comment by Wedge009** [comment]
Rebuilt with boost 1.78 and SDL 2.0.18 and can access add-ons server again. Odd.

**Comment by Wedge009** [comment]
I forgot about this issue.

Out of curiosity I retested this on both VS2019 on Windows 7 and VS2022 on Windows 10. For whatever reason I've since forgotten, I still choose to compile 1.16 branch via regular VS project, but I adopted the cmake+vcpkg builds for master. The `network_asio` invalid handle error still occurs in the former while the latter works fine. boost 1.80, SDL 2.24.2 (VS project) and SDL 2.0.22 (cmake).

Leaving this closed because it only occurs in my non-standard VS project build.
